src/coreclr/src/debug/daccess/request.cpp

```diff
@@ -18,6 +18,34 @@
 #ifdef FEATURE_COMINTEROP
 #include <comcallablewrapper.h>
 #endif // FEATURE_COMINTEROP
+#ifdef FEATURE_COMWRAPPERS
+
+// Public contract for ExternalObjectContext, keep in sync with definition in
+// interoplibinterface.cpp
```

Comment:
You could use static asserts in interoplibinterface.cpp to assert that the layout and total size of these two structs are the same (similar to DacPrivate.h) to ensure someone doesn't accidently break these without updating this code and the assert.

Just a suggestion, but at minimum I do think it's worthwhile to statically assert the size at least.

Reply:
I was going to suggest that as well. I can add that later if desired.

Reply:
I can add asserts that the fields the DAC uses are at the right offsets, but trying to assert the sizes was proving complex and ugly. I'd like to just `static_assert(sizeof(ExternalObjectContext) == sizeof(ExternalObjectContext))` but since the DAC headers are not included in comwrappers.hpp or interoplibinterface.cpp I was trying to do a static calculation of size. But with alignment padding for fields and overall padding for the structs it was getting way too complicated